lib/statsig/api_client.ex

```diff
@@ -1,55 +1,105 @@
 defmodule Statsig.APIClient do
   require Logger
-  @default_logging_api_url "https://statsigapi.net/v1/"
-  @default_config_specs_api_url "https://api.statsigcdn.com/v1/"
-
-  def download_config_specs(api_key, since_time \\ 0) do
-    base_url = get_api_url(@default_config_specs_api_url)
-    url = "#{base_url}download_config_specs/#{api_key}.json?sinceTime=#{since_time}"
-
-    result = Req.get(url: url, headers: headers(api_key))
-
-    case result do
-      {:ok, %Req.Response{status: status, body: body}} when status in 200..299 ->
-        {:ok, body}
-      {:ok, %Req.Response{status: status, body: body}} ->
-        Logger.error("HTTP error: status #{status}, body: #{inspect(body)}")
-        {:error, :http_error, status}
-      {:error, :unexpected_error, error} ->
-        Logger.error("Unexpected error: #{inspect(error)}")
-        {:error, :unexpected_error, error}
-      {:error, error} ->
-        Logger.error("Unexpected error: #{inspect(error)}")
-        {:error, :unexpected_error, error}
+  @default_logging_api_url "https://statsigapi.net/v1"
+  @default_config_specs_api_url "https://api.statsigcdn.com/v1"
+
+  def download_config_specs(since_time \\ 0) do
+    case api_key() do
+      {:ok, key} ->
+        base_url = api_url(@default_config_specs_api_url)
+
+        url =
+          URI.new!(base_url)
+          |> URI.append_path(Path.join("/download_config_specs", key <> ".json"))
+          |> URI.append_query(URI.encode_query(sinceTime: to_string(since_time)))
+          |> URI.to_string()
+
+        case Req.get(url) do
+          {:ok, %Req.Response{status: status, body: %{} = body}} when status in 200..299 ->
+            {:ok, body}
+
+          {:ok, %Req.Response{status: status} = response} when status in 200..299 ->
+            Logger.error("Invalid response format, expected map got: #{inspect(response.body)}")
+            {:error, {:invalid_response_format, response.body}}
+
+          {:ok, %Req.Response{status: status, body: body}} ->
+            Logger.error("HTTP error: status #{status}, body: #{inspect(body)}")
+            {:error, :http_error, status}
+
+          {:error, :unexpected_error, error} ->
```

Comment:
how can this happen? `Req`'s documentation doesn't mention a three-tuple unexpected error for the `get` function. 

The spec is 
```elixir
@spec get(url | keyword]() | Req.Request.t(), options :: [keyword]()) ::
  {:ok, Req.Response.t()} | {:error, Exception.t()}
  ```